test/src/helpers/school-set-up-utils.js

```diff
@@ -0,0 +1,37 @@
+const {getToken} = require('./oauth-utils');
+const {getSchoolIDBySchoolCode} = require('../services/institute-api-service');
+
+const constants = require('../config/constants');
+const restUtils = require('./rest-utils');
+import {DateTimeFormatter, LocalDate} from '@js-joda/core';
+
+const schoolSetUpUtils = {
+
+    async getSchoolDetails(schoolNumber){
+
+        const data = await getToken();
+        const token = data.access_token;
+        let schoolID = await getSchoolIDBySchoolCode(schoolNumber);
+        const url = `${constants.institute_base_url}/api/v1/institute/school/${schoolID}`;
+        return await restUtils.getData(token, url);
+    },
+    async getSchoolPrincipalDetails(schoolNumber){
+        let schoolPrincipal = '';
+        let schoolDetails = await schoolSetUpUtils.getSchoolDetails(schoolNumber);
+        for (const schoolContact of schoolDetails.contacts){
+            if(schoolContact.schoolContactTypeCode === 'PRINCIPAL'){
+                let parsedExpiryDate = null;
+                if (schoolContact.expiryDate) {
+                    parsedExpiryDate = new LocalDate.parse(schoolContact.expiryDate, DateTimeFormatter.ofPattern('uuuu-MM-dd\'T\'HH:mm:ss'));
+                }
+                if (parsedExpiryDate === null) {
+                    schoolPrincipal = schoolContact;
+                }
```

Comment:
After looking a bit more into this function, could we confirm the logic here. It sounds like if a schoolPrincipal has an expiry date we do not send it back. 

I think your tactic here was to not send back principals that are expired. 

So I think we are missing a check to see if the expiry date has less than today. You can refer to here for the logic. https://github.com/bcgov/EDUC-STUDENT-ADMIN/blob/master/frontend/src/utils/institute/status.js 

Like where your heads at :) thinking of the edge cases.